services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -1,52 +1,296 @@
-        assertEquals(routeOptions.accessToken(), updatedOptions.accessToken());
-        assertEquals(routeOptions.coordinates(), updatedOptions.coordinates());
-        assertEquals(routeOptions.user(), updatedOptions.user());
-        assertEquals(routeOptions.profile(), updatedOptions.profile());
-        assertEquals(routeOptions.geometries(), updatedOptions.geometries());
-        assertEquals(routeOptions.requestUuid(), updatedOptions.requestUuid());
-    }
-
-    private RouteOptions routeOptions() {
-        List<Point> coordinates = new ArrayList<>();
-        coordinates.add(Point.fromLngLat(1.0, 2.0));
-        coordinates.add(Point.fromLngLat(3.0, 4.0));
-
-        return RouteOptions.builder()
-                .accessToken("token")
-                .baseUrl("base_url")
-                .language("en")
-                .coordinates(coordinates)
-                .user("user")
-                .profile("profile")
-                .geometries("geometries")
-                .requestUuid("requestUuid")
-                .build();
-    }
+  @Test
+  public void toBuilder() {
+    RouteOptions routeOptions = routeOptions();
+
+    String language = "ru";
+    String url = "new_base_url";
+
+    RouteOptions updatedOptions = routeOptions.toBuilder()
+        .language(language)
+        .baseUrl(url)
+        .build();
+
+    assertEquals(language, updatedOptions.language());
+    assertEquals(url, updatedOptions.baseUrl());
+    assertEquals(routeOptions.accessToken(), updatedOptions.accessToken());
+    assertEquals(routeOptions.coordinates(), updatedOptions.coordinates());
+    assertEquals(routeOptions.user(), updatedOptions.user());
+    assertEquals(routeOptions.profile(), updatedOptions.profile());
+    assertEquals(routeOptions.geometries(), updatedOptions.geometries());
+    assertEquals(routeOptions.requestUuid(), updatedOptions.requestUuid());
+  }
+
+  @Test
+  public void radiusesString() {
+    String radiusesStr = ";5.1;;7.4;;";
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .radiuses(radiusesStr)
+        .build();
+
+    assertEquals(";5.1;;7.4", routeOptions.radiuses());
+
+    List<Double> radiuses = routeOptions.radiusesList();
+    assertEquals(4, radiuses.size());
+    assertEquals(null, radiuses.get(0));
+    assertEquals(Double.valueOf(5.1), radiuses.get(1));
+    assertEquals(null, radiuses.get(2));
+    assertEquals(Double.valueOf(7.4), radiuses.get(3));
+  }
+
+  @Test
+  public void radiusesList() {
+    List<Double> radiuses = new ArrayList<>();
+    radiuses.add(null);
+    radiuses.add(null);
+    radiuses.add(5.7);
+    radiuses.add(null);
+    radiuses.add(4.4);
+    radiuses.add(9.9);
+    radiuses.add(null);
+    radiuses.add(null);
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .radiusesList(radiuses)
+        .build();
+
+    assertEquals(radiuses, routeOptions.radiusesList());
+    assertEquals(";;5.7;;4.4;9.9", routeOptions.radiuses());
```

Comment:
```suggestion
    assertEquals(";;5.7;;4.4;9.9;;;", routeOptions.radiuses());
```

Based on the above comment, and If I'm not mistaken, this assertion should look like this.

Reply:
Yeah, that's correct 👀 https://github.com/mapbox/mapbox-java/pull/1118#discussion_r378329478

Reply:
fixed